test/entt/entity/observer.cpp

```diff
@@ -369,3 +369,34 @@ TEST(Observer, GroupCornerCase) {
     ASSERT_FALSE(add_observer.empty());
     ASSERT_TRUE(remove_observer.empty());
 }
+
+enum class CustomEntity : std::uint32_t
+{
+};
+
+
+// This could be used to add custom methods to the registry or, in case of non-public inheritance,
+// to restrict the access to the registry.
+class CustomRegistry: public entt::basic_registry<CustomEntity> {
+};
+
+template<typename Component>
+struct entt::storage_type<Component, CustomEntity>
+{
+    using type = entt::sigh_mixin<entt::basic_storage<Component, CustomEntity>, CustomRegistry>;
+};
+
+
+TEST(Observer, CustomInheritedRegistry) {
```

Comment:
To be honest, this test seems unnecessary to me. The one for the `sigh_mixin` is enough to test it. I don't see what this one brings to the table.

Reply:
It tests that the observer setup, through setting up the required signals, via the `sigh_mixin` storage, is working correctly. Can remove it though if you still think it's redundant